src/App.svelte

```diff
@@ -11,6 +13,29 @@
 
   let component;
   let params;
+  let links = [];
+
+  afterUpdate(() => {
+    const { app, ping, metric } = params;
+
+    links = [{ url: "/", name: "apps" }];
```

Comment:
I'd bet you can simplify this function a bit (i.e. removing a bunch of the if...else) by using the ternary operator (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator) and array destructuring (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment).

Reply:
To be a little more concrete about it, here's what I'd do here:

```js
const links = [
  ...[{ url: "/", name: "apps" }], 
  ...ping ? [{ url: `/apps/${app}/pings/${ping}/`, name: ping }] : [],
  ...metric ? [{ url: `/apps/${app}/metrics/${metric}/`, name: metric }] : [],
];
```

(this example doesn't include BigQuery, leaving that as an exercise for you!)